src/sentry/replays/selector.py

```diff
@@ -0,0 +1,85 @@
+from typing import List, Optional, Union
+
+from cssselect import Attrib, Class, Element, Hash, Selector
+
+SelectorType = Union[Attrib, Class, Element, Hash]
+
+
+class QueryType:
```

Comment:
does it make sense to use a dataclass here?

Reply:
Markus told me to avoid them because they're slow.  But yes I think they would make sense here all else being equal.

Reply:
I think performance wise this is equivalent to a dataclass (believe he was saying to use a TypedDict). but I don't think that matters here in this endpoint as we're only creating one, only in the consumer where we could create N number of them.